docs/README.md

```diff
@@ -0,0 +1,613 @@
+    - [Iterators](#iterators)
+        - [CircularIterator](#circulariterator)
+        - [CompositeIterator](#compositeiterator)
+        - [MergedBufferedIterator](#mergedbufferediterator)
+- [Encryption](#encryption)
+- [Hashing](#hashing)
+- [IO](#io)
+    - [FileDescriptor](#filedescriptor)
+    - [ResourceUtil](#resourceutil)
+- [JSON](#json)
+- [Profiling](#profiling)
+    - [CpuSampler](#cpusampler)
+    - [SimpleJmx](#simplejmx)
+- [Time](#time)
+- [TestKit](#testkit)
+
+## Core
+
+To use it in an existing SBT project, add the following dependency to your `build.sbt`:
+
+```scala
+libraryDependencies += "com.velocidi" %% "apso-core" % "@VERSION@"
+```
+
+### Config
+
+Apso provides methods to ease working with Typesafe's [config](https://github.com/typesafehub/config).
+
+#### LazyConfigFactory
+
+The `LazyConfigFactory` object provides static methods for creating `Config` instances in a lazy way. The lazy way refers to the variable loading process. The usual process loads variables in config files eagerly (i.e. the path needs to be defined in the same file it is refered to). The loading process provided by `LazyConfigFactory` loads and merges all configuration files and only then resolves variables. This loading process introduces a third file (beyond the default ones - `application.conf` and `reference.conf`): `overrides.conf`. This file has priority over the `application.conf` file and can be used to specify keys that should always be overriden, e.g. by environment variables.
+
+### HTTP
+
+Apso provides a tiny wrapper for [Dispatch](http://dispatch.databinder.net/) with synchronous operations. It's called `W`, and the following shows some sample usage:
+
+```scala mdoc:compile-only
+import com.velocidi.apso.http.W
+
+W.get("http://www.google.com/").getStatus
+// res0: Int = 302
+
+W.post("http://www.google.com/", "").getStatus
+// res1: Int = 405
+
+W.put("http://www.google.com/", "").getStatus
+// res2: Int = 405
+
+W.delete("http://www.google.com/").getStatus
+// res3: Int = 405
+
+W.head("http://www.google.com/").getStatus
+// res4: Int = 302
+```
+
+The POST and PUT methods can also receive the body as `JSON` (of [circe](https://github.com/circe/circe)), which adds the `Content-type` header accordingly.
+
+### Geo
+
+The `Geo` object provides methods to compute distances in kilometers between two points on the planet Earth, calculated using the spherical [law of cosines](https://en.wikipedia.org/wiki/Great-circle_distance#Formulas). Coordinates are represented by a pair of `Double` for latitude and longitude.
+
+```scala mdoc:reset
+import com.velocidi.apso.Geo
+
+Geo.distance((41.1617609, -8.6024716), (41.1763745, -8.5964861))
+```
+
+You can also have the distance function curried if you are computing distances from a fixed point:
+
+```scala mdoc
+val distFromOffice = Geo.distanceFrom((41.1617609, -8.6024716))
+
+distFromOffice((41.1763745, -8.5964861))
+
+distFromOffice((38.7223032, -9.1414664))
+```
```

Comment:
We should wrap the `distFromOffice` declaration in a silent block to avoid having a change in the docs with an instance that doesn't have a stable `toString`.

```diff
diff --git a/docs/README.md b/docs/README.md
index 0cb0733..ec19dd2 100644
--- a/docs/README.md
+++ b/docs/README.md
@@ -119,9 +119,11 @@ Geo.distance((41.1617609, -8.6024716), (41.1763745, -8.5964861))
 
 You can also have the distance function curried if you are computing distances from a fixed point:
 
-```scala mdoc
+```scala mdoc:silent
 val distFromOffice = Geo.distanceFrom((41.1617609, -8.6024716))
+```
 
+```scala mdoc
 distFromOffice((41.1763745, -8.5964861))
 
 distFromOffice((38.7223032, -9.1414664))
```

Reply:
On db85c699bd224347d2c3365383b4a0c6f6014dfd, I've made the suggested changes (and some more) to avoid the unstable output.